geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/MessageDispatcher.java

```diff
@@ -193,6 +194,15 @@ public boolean isWaitingForReAuthentication() {
     return waitForReAuthenticationStartTime > 0;
   }
 
+  private volatile boolean subjectUpdated = false;
+
+  public void notifyReAuthentication() {
+    synchronized (re_auth_lock) {
+      subjectUpdated = true;
+      re_auth_lock.notify();
```

Comment:
You always want to use `notifyAll()` instead of `notify()`. I don't remember exact reasons off the top of my head but I probably picked it up most recently from Java Concurrency in Practice by Brian Goetz.

Reply:
done